visual-java/src/main/java/com/saucelabs/visual/graphql/GraphQLClient.java

```diff
@@ -63,18 +69,52 @@ public GraphQLClient(
     this.objectMapper = objectMapper;
   }
 
+  /** Inject the headers & configuration options for interacting with the GraphQL API. */
+  private void configureRequest(HttpRequestBase request) {
+    request.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + authentication);
+    request.setHeader(
+        HttpHeaders.USER_AGENT,
+        "sauce-visual-java/" + ArtifactVersion.getArtifactVersion().orElse("unknown"));
+    request.setConfig(requestConfig);
+  }
+
+  /** Grab our DOM Capture script from the API if not already cached locally. */
+  public String getDomCaptureScript() {
+    if (GraphQLClient.domCaptureScript != null) {
+      return GraphQLClient.domCaptureScript;
+    }
+
+    String url = this.staticUrl + "browser-scripts/dom-capture.js";
+    try {
+      HttpGet request = new HttpGet(url);
+      configureRequest(request);
+      HttpResponse response = this.client.execute(request);
+      int status = response.getStatusLine().getStatusCode();
+      HttpEntity entity = response.getEntity();
+      String responseString = EntityUtils.toString(entity, "UTF-8");
+
+      if (status != 200) {
+        throw new VisualApiException("Unexpected status code: " + status + " " + responseString);
+      }
+
+      GraphQLClient.domCaptureScript = responseString;
+      return responseString;
+    } catch (IOException e) {
+      throw new VisualApiException(e.getMessage(), e);
+    }
+  }
+
   /**
-   * Uploads an image/png file to Sauce's visual storage.
+   * Uploads a file to Sauce's visual storage.
    *
    * @param uri The URL retrieved from the createSnapshotUpload mutation.
-   * @param image An image byte array, retrieved from the selenium driver's getScreenshotAs()
-   *     method.
+   * @param content A byte array containing the contents to upload to Visual storage.
    */
-  public void upload(String uri, byte[] image) throws VisualApiException {
+  public void upload(String uri, byte[] content, String mimeType) throws VisualApiException {
```

Comment:
As this is a public API, maybe `mimeType` should be optional and set to `"image/png"` by default? Or, add an overload without `mimeType` and set it as deprecated?

Reply:
This only exists on top of the `java-local` feature branch which hasn't been released yet, and even then we anticipate limited usage. Not sure it's worth adding / maintaining the override for it